opm/common/OpmLog/KeywordLocation.hpp

```diff
@@ -20,10 +20,25 @@
 #ifndef KEYWORD_LOCATION_HPP
 #define KEYWORD_LOCATION_HPP
 
+#include <string>
+
 namespace Opm {
 
 class KeywordLocation {
 public:
+    /*
+      Observe that many error messages whcih should print out the name of the
+      problem keyword along with the location {} placeholders can be used. The
+      convention is:
+
+         {0} -> keyword
+         {1} -> filename
+         {2} -> lineno
```

Comment:
What about using named arguments so that we can use `{keyword}` etc. instead? This could be set up in the what() function discussed in #1942 for example, using `fmt::arg()`.

Reply:
The reason I went with `{0}, {1} and {2}` was that this is standard format functionality; I was not aware of `fmt::arg()` but that looks a good way to get named arguments. I will look into it.

Reply:
> What about using named arguments so that we can use `{keyword}` etc. instead?

For what it's worth, I'm using this approach in #1938, specifically in [`TableManager.cpp`](https://github.com/OPM/opm-common/blob/ca2cbd0e69fe547a5d010c77047b5af66cc653af/src/opm/parser/eclipse/EclipseState/Tables/TableManager.cpp#L1238-L1250).  You don't even need to (manually) construct `fmt::arg` instances.  If you
```C++
#include <fmt/format.h>
```
(as opposed to `<fmt/core.h>`) you get the user-defined literal `operator "_a()` in namespace `fmt::literals` (needs a `using` directive in scope) which is, I think, as close to real named arguments as it is possible to get.